middle_end/flambda2/term_basics/alloc_mode.mli

```diff
@@ -12,26 +12,28 @@
 (*                                                                        *)
 (**************************************************************************)
 
-type t = private
-  | Heap  (** Normal allocation on the OCaml heap. *)
-  | Local  (** Allocation on the local allocation stack. *)
+module For_types : sig
+  type t = private
+    | Heap  (** Normal allocation on the OCaml heap. *)
+    | Local  (** Allocation on the local allocation stack. *)
+    | Heap_or_local  (** Allocation with unknown location *)
 
-type without_region = t
+  val print : Format.formatter -> t -> unit
 
-val print : Format.formatter -> t -> unit
+  val compare : t -> t -> int
 
-val compare : t -> t -> int
+  val heap : t
 
-val heap : t
+  val local : t
 
-(** Returns [Heap] if stack allocation is disabled! *)
-val local : unit -> t
+  val unknown : t
 
-val from_lambda : Lambda.alloc_mode -> t
+  val from_lambda : Lambda.alloc_mode -> t
 
-val to_lambda : t -> Lambda.alloc_mode
+  val to_lambda : t -> Lambda.alloc_mode
+end
 
-module With_region : sig
+module For_allocations : sig
   type t = private
     | Heap  (** Normal allocation on the OCaml heap. *)
     | Local of { region : Variable.t }
```

Comment:
This should probably have a comment

Reply:
It has, it's on the line after (not shown in the diff, I guess)